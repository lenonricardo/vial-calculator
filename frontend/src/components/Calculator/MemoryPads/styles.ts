import styled from 'styled-components'
import { Pad } from '../NumberPads/styles'

export const MemoryWrapper = styled.div`
  display: flex;
  flex-direction: column;
  transform: translateY(var(--pad-width));
`
export const SecondaryPad = styled(Pad)`
  background: var(--secondary-operations);

  &:hover {
    background: var(--secondary-operations);
  }
`
