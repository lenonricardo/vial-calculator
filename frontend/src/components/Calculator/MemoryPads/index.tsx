import { MemoryWrapper, SecondaryPad } from './styles'

interface MemoryPadsProps {
  onInput: (input: string) => void
}

enum MemoryPadsEnum {
  MEMORY_PLUS = 'M+',
  MEMORY_MINUS = 'M-',
  MEMORY_RECALL = 'MR',
  MEMORY_CLEAR = 'MC'
}

export default function MemoryPads({ onInput }: MemoryPadsProps) {
  const memoryPads: MemoryPadsEnum[] = Object.values(MemoryPadsEnum)

  const handleOperationClick = (operation: string) => {
    onInput(operation)
  }

  return (
    <MemoryWrapper>
      {memoryPads.map((label) => (
        <SecondaryPad key={label}>{label}</SecondaryPad>
      ))}
    </MemoryWrapper>
  )
}
