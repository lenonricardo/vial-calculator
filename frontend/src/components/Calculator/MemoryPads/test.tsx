import { render } from '@testing-library/react'

import MemoryPads from '.'

describe('<MemoryPads />', () => {
  it('should render the component', () => {
    const mockOnInput = jest.fn()

    const { container } = render(<MemoryPads onInput={mockOnInput} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
