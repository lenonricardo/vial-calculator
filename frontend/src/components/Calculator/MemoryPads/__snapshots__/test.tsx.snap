// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoryPads /> should render the component 1`] = `
.c1 {
  width: var(--pad-width);
  height: var(--pad-height);
  color: var(--font-pads);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: var(--default-font-size);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transform: translateY(var(--pad-width));
  -ms-transform: translateY(var(--pad-width));
  transform: translateY(var(--pad-width));
}

.c2 {
  background-color: var(--secondary-operations);
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    M+
  </div>
  <div
    class="c1 c2"
  >
    M-
  </div>
  <div
    class="c1 c2"
  >
    MR
  </div>
  <div
    class="c1 c2"
  >
    MC
  </div>
</div>
`;
